app/models/cta.rb

```diff
@@ -0,0 +1,36 @@
+class CTA < ApplicationRecord
+  CTA_TYPES = { onsite: 1, phone: 2 }
+
+  belongs_to :contact, optional: true
+  belongs_to :location, optional: true
+  belongs_to :call_script, optional: true
+
+  validates_presence_of :title, :description, :website, :start_at, :cta_type
+  validate :validate_unique_cta, on: :create
+
+  scope :upcoming, -> { where("start_at >= ?", Date.today).order("start_at") }
```

Comment:
I'd use `DateTime.current` instead of `Date.today` here to avoid using the server's timezone instead of the application's timezone (which we should set to `utc` in `application.rb`).

Also we should consider indexing the `start_at` column to avoid a full table scan with this scope.